### DISCUSSION TITLE
Using the JS library to implement a query with a one-to-many inner join

### DISCUSSION THREAD

**Discussion by patroucheva** [supabase/supabase#28860] [Questions]
I'm trying to implement a query using the JavaScript library.

My example setup is quite simple, say I have two tables:

**Children**

```
+---+--------+-------------+------------+
| id| name   | parent_a_id | parent_b_id|
+---+--------+-------------+------------+
| 1 | Amy    | 1           | 3          |
+---+--------+-------------+------------+
| 2 | Amanda | 1           | 3          |
+---+--------+-------------+------------+
| 3 | Alex   | 1           | 2          |
+---+--------+-------------+------------+
```
**Parents**
```
+---+--------+
| id| name   |
+---+--------+
| 1 | John   |
+---+--------+
| 2 | Jessica|
+---+--------+
| 3 | Juliana|
+---+--------+
```
My goal: Select all children who's parent_b is named Jessica. The result should only be Alex.

**My current approach:**

```
    let query = supabase
                .from('children')
                .select('
                        *,
                        parent_a:parents!inner(parent_a_id)(name),
                        parent_b:parents!inner(parent_b_id)(name)
                 ')
                 .eq('parent_b.name', 'Jessica');
```

This does not work. I get the following error due to the one-to-many relationship:

Could not embed because more than one relationship was found for 'Children' and 'Parents'

Does anyone have an approach to doing this properly with the JavaScript library? Thank you in advance!

**Comment by GaryAustin1** [comment]
Usually it gives you suggestions.
But you can specify the exact column to use like ` parent_a:parents!parent_a_id!inner(name),` I believe.
https://docs.postgrest.org/en/v12/references/api/resource_embedding.html#multiple-one-to-many
